### ISSUE TITLE
[Bug] KeyError: 'avg_loss_1' crash when training model

### ISSUE THREAD

**Issue by T145** [coqui-ai/TTS#2862]
### Describe the bug

Occurs when running the script below.

### To Reproduce

Run the following script. I've tried it w/ the `LJSpeech-1.1` dataset and the same error occurs, so you can sub in the desired dataset.

```python
import os
from TTS.tts.configs.shared_configs import BaseAudioConfig, BaseDatasetConfig
from TTS.tts.configs.glow_tts_config import GlowTTSConfig
from TTS.utils.audio import AudioProcessor
from TTS.tts.utils.text.tokenizer import TTSTokenizer
from TTS.tts.datasets import load_tts_samples
from TTS.tts.models.glow_tts import GlowTTS
from trainer import Trainer, TrainerArgs


def main():
	output_path = os.path.join(os.getcwd(), 'runtime')

	if not os.path.exists(output_path):
		os.makedirs(output_path)

	dataset = 'FerdinandTTS'
	dataset_config = BaseDatasetConfig(
		formatter='ljspeech',
		meta_file_train='metadata.csv',
		path=os.path.join(os.getcwd(), 'datasets', dataset),
		dataset_name=dataset,
		language='en-us',
		phonemizer='espeak',
	)

	audio_config = BaseAudioConfig(
		sample_rate=48000
	)

	config = GlowTTSConfig(
		batch_size=64, #32, # https://github.com/coqui-ai/TTS/issues/1447#issuecomment-1083100386
		eval_batch_size=16,
		num_loader_workers=4,
		num_eval_loader_workers=4,
		run_eval=False, #True,
		test_delay_epochs=10, #-1,
		epochs=100,
		text_cleaner="phoneme_cleaners",
		use_phonemes=True,
		phoneme_language="en-us",
		phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
		print_step=25,
		print_eval=False,
		mixed_precision=False, # https://github.com/coqui-ai/TTS/issues/918
		output_path=output_path,
		datasets=[dataset_config],
		save_step=25000, #1000,
		# CUSTOM
		audio=audio_config,
		phonemizer='espeak',
		#eval_split_size=10,
		target_loss='G_avg_loss',
		#max_audio_len=1024*3,
		#max_text_len=1000,
		eval_split_max_size=0.01,
		eval_split_size=0.006666666666666667
	)

	ap = AudioProcessor.init_from_config(config)
	tokenizer, config = TTSTokenizer.init_from_config(config)
	train_samples, eval_samples = load_tts_samples(
		dataset_config,
		eval_split=True,
		eval_split_max_size=config.eval_split_max_size,
		eval_split_size=config.eval_split_size
	)
	model = GlowTTS(config, ap, tokenizer, speaker_manager=None)
	trainer_args = TrainerArgs(
		gpu=0, # Prevents having to set CUDA_VISIBLE_DEVICES=0
		use_accelerate=True
	)
	trainer = Trainer(
		trainer_args, config, output_path, model=model, train_samples=train_samples, eval_samples=eval_samples
	)

	trainer.fit()


if __name__ == '__main__':
	main()
```

### Expected behavior

To produce a functional model.

### Logs

```shell
(cudatest) PS {project path}> python .\test_train.py
 > Setting up Audio Processor...
 | > sample_rate:48000
 | > resample:False
 | > num_mels:80
 | > log_func:np.log10
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.5
 | > preemphasis:0.0
 | > griffin_lim_iters:60
 | > signal_norm:True
 | > symmetric_norm:True
 | > mel_fmin:0
 | > mel_fmax:None
 | > pitch_fmin:1.0
 | > pitch_fmax:640.0
 | > spec_gain:20.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:True
 | > trim_db:45
 | > do_sound_norm:False
 | > do_amp_to_db_linear:True
 | > do_amp_to_db_mel:True
 | > do_rms_norm:False
 | > db_level:None
 | > stats_path:None
 | > base:10
 | > hop_length:256
 | > win_length:1024
 | > Found 150 files in {project path}\datasets\FerdinandTTS
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
 > Training Environment:
 | > Backend: Accelerate
 | > Mixed precision: False
 | > Precision: float32
 | > Current device: 0
 | > Num. of GPUs: 1
 | > Num. of CPUs: 12
 | > Num. of Torch Threads: 6
 | > Torch seed: 54321
 | > Torch CUDNN: True
 | > Torch CUDNN deterministic: False
 | > Torch CUDNN benchmark: False
 | > Torch TF32 MatMul: False
 > Start Tensorboard: tensorboard --logdir={project path}\runtime\run-August-11-2023_08+00PM-0000000

 > Model has 28610257 parameters

 > EPOCH: 0/100
 --> {project path}\runtime\run-August-11-2023_08+00PM-0000000


> DataLoader initialization
| > Tokenizer:
        | > add_blank: False
        | > use_eos_bos: False
        | > use_phonemes: True
        | > phonemizer:
                | > phoneme language: en-us
                | > phoneme backend: espeak
| > Number of instances : 149
 | > Preprocessing samples
 | > Max text length: 140
 | > Min text length: 4
 | > Avg text length: 71.6510067114094
 |
 | > Max audio length: 422066.0
 | > Min audio length: 18686.0
 | > Avg audio length: 210688.61744966442
 | > Num. instances discarded samples: 0
 | > Batch group size: 0.

 > TRAINING (2023-08-11 20:00:38)

   --> TIME: 2023-08-11 20:01:07 -- STEP: 0/3 -- GLOBAL_STEP: 0
     | > current_lr: 2.5e-07
     | > step_time: 2.9498  (2.9498186111450195)
     | > loader_time: 25.6026  (25.602627754211426)

 [!] `train_step()` retuned `None` outputs. Skipping training step.
 [!] `train_step()` retuned `None` outputs. Skipping training step.
 [!] `train_step()` retuned `None` outputs. Skipping training step.

  --> EVAL PERFORMANCE
     | > avg_loader_time: 29.987560629844666 (+0)
     | > avg_step_time: 1.201514482498169 (+0)

Traceback (most recent call last):
  File "{python env}\cudatest\Lib\site-packages\trainer\trainer.py", line 1806, in fit
    self._fit()
  File "{python env}\cudatest\Lib\site-packages\trainer\trainer.py", line 1769, in _fit
    self.save_best_model()
  File "{python env}\cudatest\Lib\site-packages\trainer\utils\distributed.py", line 35, in wrapped_fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "{python env}\cudatest\Lib\site-packages\trainer\trainer.py", line 1886, in save_best_model
    target_loss_dict = self._pick_target_avg_loss(self.keep_avg_eval if self.keep_avg_eval else self.keep_avg_train)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "{python env}\cudatest\Lib\site-packages\trainer\trainer.py", line 2103, in _pick_target_avg_loss
    target_loss = keep_avg_target["avg_loss_1"]
                  ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
  File "{python env}\cudatest\Lib\site-packages\trainer\generic_utils.py", line 119, in __getitem__
    return self.avg_values[key]
           ~~~~~~~~~~~~~~~^^^^^
KeyError: 'avg_loss_1'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "{project path}\test_train.py", line 80, in <module>
    main()
  File "{project path}\test_train.py", line 76, in main
    trainer.fit()
  File "{python env}\cudatest\Lib\site-packages\trainer\trainer.py", line 1833, in fit
    remove_experiment_folder(self.output_path)
  File "{python env}\cudatest\Lib\site-packages\trainer\generic_utils.py", line 77, in remove_experiment_folder
    fs.rm(experiment_path, recursive=True)
  File "{python env}\cudatest\Lib\site-packages\fsspec\implementations\local.py", line 172, in rm
    shutil.rmtree(p)
  File "{python env}\cudatest\Lib\shutil.py", line 759, in rmtree
    return _rmtree_unsafe(path, onerror)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "{python env}\cudatest\Lib\shutil.py", line 622, in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
  File "{python env}\cudatest\Lib\shutil.py", line 620, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: '{project path}/runtime/run-August-11-2023_08+00PM-0000000\\trainer_0_log.txt'
```

### Environment

```shell
{
    "CUDA": {
        "GPU": [
            "NVIDIA GeForce GTX 1050 Ti"
        ],
        "available": true,
        "version": "11.7"
    },
    "Packages": {
        "PyTorch_debug": false,
        "PyTorch_version": "2.0.1",
        "TTS": "0.16.2",
        "numpy": "1.24.3"
    },
    "System": {
        "OS": "Windows",
        "architecture": [
            "64bit",
            "WindowsPE"
        ],
        "processor": "Intel64 Family 6 Model 158 Stepping 10, GenuineIntel",
        "python": "3.11.4",
        "version": "10.0.17763"
    }
}
```

### Additional context

N/A

**Comment by Myst1ck** [comment]
Got the same problem too, for some reason `keep_avg_target.avg_values` is an empty dictionary...
Tried to fix it with #1608 but didn't fix it for me.

**Comment by T145** [comment]
The latest dev changes and enabling virtualization on my CPU got everything working! Not sure if those are directly correlated but that's what happened.
